lib/internal/process.js

```diff
@@ -78,14 +78,19 @@ function setup_hrtime() {
     _hrtime(hrValues);
 
     if (time !== undefined) {
-      if (Array.isArray(time) && time.length === 2) {
-        const sec = (hrValues[0] * 0x100000000 + hrValues[1]) - time[0];
-        const nsec = hrValues[2] - time[1];
-        const needsBorrow = nsec < 0;
-        return [needsBorrow ? sec - 1 : sec, needsBorrow ? nsec + 1e9 : nsec];
+      if (!Array.isArray(time)) {
+        throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'time', 'Array',
+                                   time);
       }
-      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
-                                 'process.hrtime()', 'Array');
+      if (time.length !== 2) {
+        throw new errors.TypeError('ERR_INVALID_ARRAY_LENGTH', 'time', 2,
```

Comment:
I wasn't sure about this either. `RangeError` sounds logical at first considering that the length is out of the valid range (even though that range only includes a single value), but on the other hand, the array is used like a mathematical tuple and I would say that for tuples, the number of elements is part of the type. IMO `RangeError` is better when it is either about a numeric value passed to a function or if there is actually a specific range of allowed implicit values such as the array length (which permits more than a single value). Both `TypeError` and `RangeError` are fine by me, but I would like some other opinions on this.